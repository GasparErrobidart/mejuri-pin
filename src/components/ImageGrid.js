import styled from 'styled-components'

const gap = "40px"

const ImageGrid = styled.div`
  display:grid;
  grid-template-columns: repeat( auto-fit, minmax(200px,1fr) );
  grid-gap: ${gap};
  margin-top: ${gap};
  margin-bottom: ${gap};
`;

export default ImageGrid;
