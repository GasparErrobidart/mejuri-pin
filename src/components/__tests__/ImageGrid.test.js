import '@testing-library/jest-dom'
import 'jest-styled-components'

import React         from 'react'
import { cleanup }   from '@testing-library/react'
import renderer      from 'react-test-renderer'

import ImageGrid   from '../ImageGrid'


afterEach(cleanup)

test('Renders "ImageGrid" component', () => {
  const testContent     = Array(6)
                            .fill(null)
                            .map((d,i) => <img key={i} src="/placeholder.png" alt="Test image" />)
  const component       = renderer.create(<ImageGrid>{testContent}</ImageGrid>);
  const tree            = component.toJSON()
  expect(tree).toMatchSnapshot();
});
