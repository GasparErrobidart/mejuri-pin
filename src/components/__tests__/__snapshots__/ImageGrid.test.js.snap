// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders "ImageGrid" component 1`] = `
.c0 {
  display: grid;
  grid-template-columns: repeat( auto-fit,minmax(20vh,40vh) );
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  grid-gap: 20px;
  margin: 30px 10px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-align-content: start;
  -ms-flex-line-pack: start;
  align-content: start;
}

.c0 > * {
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
}

@media (min-width:768px) {
  .c0 {
    margin: 40px;
    grid-gap: 40px;
  }
}

<div
  className="c0"
>
  <img
    alt="Test image"
    src="/placeholder.png"
  />
  <img
    alt="Test image"
    src="/placeholder.png"
  />
  <img
    alt="Test image"
    src="/placeholder.png"
  />
  <img
    alt="Test image"
    src="/placeholder.png"
  />
  <img
    alt="Test image"
    src="/placeholder.png"
  />
  <img
    alt="Test image"
    src="/placeholder.png"
  />
</div>
`;
